File: src/test/java/boj/problems/step9/No11651Test.java
package boj.problems.step9;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.io.StringReader;
import java.io.StringWriter;
import java.util.stream.Collectors;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class No11651Test {
    File path = new File(".");
    String input = path.getAbsolutePath() + "/src/test/java/boj/problems/step9/No11651_input.txt";
    String output = path.getAbsolutePath() + "/src/test/java/boj/problems/step9/No11651_output.txt";

    @Test
    @DisplayName("좌표 정렬하기 2")
    void 좌표_정렬하기_2_테스트() throws IOException {
        System.out.println("좌표 정렬하기 2 : https://www.acmicpc.net/problem/11651");

        BufferedReader br_given = new BufferedReader(new FileReader(input));
        BufferedReader br_want = new BufferedReader(new FileReader(output));
        StringWriter sw = new StringWriter();
        BufferedWriter bw_got = new BufferedWriter(sw);

        No11651.solve(br_given, bw_got);

        bw_got.close();

        BufferedReader br_got = new BufferedReader(new StringReader(sw.getBuffer().toString()));

        String got = br_got.lines().collect(Collectors.joining());
        String want = br_want.lines().collect(Collectors.joining());

        assertEquals(want, got);
    }
}

File: src/test/kotlin/boj/problems/step13/No11651Test.kt
package boj.problems.step13

import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe
import java.io.BufferedReader
import java.io.StringReader

class No11651Test : StringSpec({
    "좌표 정렬하기 2 : https://www.acmicpc.net/problem/11651" {
        val input = """
            5
            0 4
            1 2
            1 -1
            2 2
            3 3
        """.trimIndent()
        val expected = """
            1 -1
            1 2
            2 2
            3 3
            0 4
        """.trimIndent()

        val actual = No11651().solve(BufferedReader(StringReader(input)))

        actual shouldBe expected
    }
})
